src/dali/abstract_pair_converter_plugin.py

```diff
@@ -88,6 +93,45 @@ def get_or_set_vertex(self, name: str) -> Vertex[ValueType]:
         self.add_vertex(new_vertex)
         return new_vertex
 
+    def is_optimized(self, asset: str) -> bool:
+        return asset in self.__optimized_assets
+
+    # Optimized asset is the asset we are optimizing during this cloning.
+    # Optimized assets are tracked to prevent redundant cloning.
+    # Negative weights will get deleted.
+    def clone_with_optimization(self, optimized_asset: str, neighbor_weights: Dict[str, float]) -> "MappedGraph[ValueType]":
+        if optimized_asset in self.__optimized_assets:
+            raise RP2ValueError(
+                "You are trying to optimize an asset that has already been optimized."
+                "Use is_optimized() to check if an asset has already been optimized before cloning."
+            )
+
+        cloned_optimized_assets: Dict[str, None] = self.__optimized_assets
+        cloned_optimized_assets[optimized_asset] = None
+        cloned_mapped_graph: MappedGraph[ValueType] = MappedGraph(optimized_assets=cloned_optimized_assets)
+
+        for original_vertex in super().vertexes:
+            # Add existing neighbors
+            for neighbor in original_vertex.neighbors:
+                neighbor_weight: float
+                if original_vertex.name == optimized_asset:
```

Comment:
I'm not too clear what this bit is doing (it may be caused by my original uncertainty with the "optimized" concept)

Reply:
We are only going to be optimizing one asset at a time. So, we need to change the weights of its neighbors (ie. quote assets) only if they were passed to the function. If they were not passed to the function, we keep the original weights.